org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/serializer/ChangeSerializerTest.xtend

```diff
@@ -506,6 +506,25 @@ class ChangeSerializerTest {
 	}
 
 	@Test
+	def void testAddElementsWithEmptyLineUnix() {
+		val oldLineSeparator = System.getProperty("line.separator")
+		System.setProperty("line.separator", "\n")
+		try {
+			testAddElementsWithEmptyLine()
+		} finally {
+			System.setProperty("line.separator", oldLineSeparator)
+		}	
+	}
+	@Test
+	def void testAddElementsWithEmptyLineWindows() {
+		val oldLineSeparator = System.getProperty("line.separator")
+		System.setProperty("line.separator", "\r\n")
```

Comment:
Naive attempt at testing in unix & windows modes, it seems, however, that the formatter preferences are cached and hence the first value wins for all tests. Ergo, running tests individually works as designed, running the suite still only tests one mode.
Feel free to suggest a way how individual tests can use different line separator settings ...

Reply:
i also observed this problem. either there is a bug with caching or we should use unix endings all over the place.

Reply:
the caching problem is also there when the cached value is guessed by looking at a file and this file has non-typical endings